Request/ParamConverter/DoctrineParamConverter.php

```diff
@@ -162,16 +170,42 @@ protected function findOneBy($class, Request $request, $options)
         }
 
         if (isset($options['repository_method'])) {
-            $method = $options['repository_method'];
+            $repositoryMethod = $options['repository_method'];
         } else {
-            $method = 'findOneBy';
+            $repositoryMethod = 'findOneBy';
         }
 
         try {
-            return $em->getRepository($class)->$method($criteria);
+            if ($mapMethodSignature) {
+                $result = $this->findDataByMapMethodSignature($em, $class, $repositoryMethod, $criteria);
+            } else {
+                $result = $em->getRepository($class)->$repositoryMethod($criteria);
+            }
         } catch (NoResultException $e) {
-            return;
+            $result = null;
+        }
+
+        return $result;
+    }
+
+    private function findDataByMapMethodSignature($em, $class, $repositoryMethod, $criteria)
+    {
+        $repository = $em->getRepository($class);
+        $repositoryClass = get_class($repository);
```

Comment:
You can remove this and use `$repository` directly as an argument to the Reflection constructors.